rustls/src/client/client_conn.rs

```diff
@@ -280,7 +281,15 @@ impl ClientConfig {
     /// Return true if connections made with this `ClientConfig` will
     /// operate in FIPS mode.
     pub fn fips(&self) -> bool {
-        self.provider.fips()
+        #[cfg(feature = "tls12")]
```

Comment:
If we're going to make the config-side `fips()` depend on the config, should we make the `CryptoProvider::fips()` `pub(crate)` to make it harder to accidentally rely on the `CryptoProvider::fips()` where that is not sufficient?

Reply:
I think -- I hope -- that there is not much confusion in semantics between `CryptoProvider::fips()` and `Client/ServerConfig::fips()`, even if they were functionally the same up until this commit. Will extend the docs though.

Reply:
Maybe the distinction needs to be explained in `CryptoProvider::fips()`? That's where it's risky, if you use that `fips()` but fail to check the config.

Reply:
That seems like a good idea to me

Reply:
I have added some clarifying text to `CryptoProvider::fips()` -- please check.

Reply:
LGTM